neo/app/controllers/Application.scala

```diff
@@ -151,12 +153,40 @@ object Application{
   }
 
   def findCity(neLat: Double, swLat: Double, neLng: Double, swLng: Double, cities: List[JsValue]) =  {
-    //TODO: Do binary search
-    val citiesWithinBoundary = cities.filter{
-      city =>
-        (city \ CentroidLatitude).as[Double] <= neLat && (city \ CentroidLatitude).as[Double] >= swLat.toDouble && (city \ CentroidLongitude).as[Double] <= neLng.toDouble && (city \ CentroidLongitude).as[Double] >= swLng.toDouble
+    /*
```

Comment:
Every function should explicitly declare the return type.
`def findCity(neLat: Double, swLat: Double, neLng: Double, swLng: Double, cities: List[JsValue])` ==> 
`def findCity(neLat: Double, swLat: Double, neLng: Double, swLng: Double, cities: List[JsValue]) : JsArray`